### PR TITLE
feat(core): Support minor levels in n8nNodesApiVersion (no-changelog)

### PR DESCRIPTION
**Pull request by garritfra**
## Summary

`n8nNodesApiVersion` was assumed to hold major versions only, so the
compatibility gate accepted positive integers and nothing else. A package that
declares `3.1` was classified as **malformed metadata** on every instance,
including `3.x` — the user was told the package is corrupt instead of being told
which n8n version it needs.

This PR makes a level `<major>.<minor>` through the whole chain:

- `parseNodesApiLevel()` (new, exported from `n8n-workflow`) parses `3`, `"3"`
  and `"3.1"` into `[major, minor]`. `checkNodesApiVersion()` compares major
  first, then minor.
- `N8N_NODES_API_VERSION` becomes a level string (`'1'` on master, `'3.0'` on
  `3.x`). Bump the minor in the release that adds a node-authoring API inside a
  major.
- The install, update, startup and reinstall guards, the error metadata
  (`requiredNodesApiVersion` / `supportedNodesApiVersion` are now strings), the
  frontend mirror of that metadata, and the `maxN8nNodesApiVersion` catalog
  request parameter all carry the level unchanged instead of a number.
- The `n8n-object-validation` lint rule accepts a level, so `n8n-node lint` no
  longer fails a package that declares a minor.

**A minor level must be a string.** A JSON number cannot carry one: `3.10`
parses as `3.1`, so minor 10 is unrepresentable, and `3.9 > 3.10` numerically.
Numbers stay valid in their integer form (`3` means `3.0`), so every published
package keeps working and the catalog request parameter stays wire-compatible
with instances already in the field. A fractional number is rejected rather than
guessed at.

Behaviour on old instances is unchanged and still safe: the shipped integer-only
guard reads `"3.1"` as malformed and blocks the package. Only the message is
less precise there, and that cannot be fixed retroactively (CE-2571 was
canceled).

Follow-up work in other repos is needed before the first minor level is used —
see the Linear ticket: the Strapi catalog stores the level in a SQL-filtered
scalar column that mis-orders minors, and the promotion pipeline and Verdaccio
allowlist key release lines by API level.

## How to test

1. `cd packages/workflow && pnpm test test/nodes-api-version.test.ts` — parse
   and boundary cases, including `"3.10"` vs `3.1`.
2. `cd packages/core && pnpm test src/nodes-loader/__tests__/scan-directory-for-packages` —
   startup guard skips a package one minor above the supported level.
3. `cd packages/cli && pnpm test src/modules/community-packages` — install,
   update and reinstall guards.
4. `cd packages/@n8n/eslint-plugin-community-nodes && pnpm test src/rules/n8n-object-validation.test.ts` —
   `"3.1"` and `"3.10"` are valid, `3.1` as a number is not.
5. Manually: put a package declaring `"n8nNodesApiVersion": "3.1"` in
   `~/.n8n/nodes`, start n8n, and check the startup warning names the declared
   and supported levels. Installing it from Settings > Community nodes must be
   rejected with the "not compatible with this n8n version" copy, not the
   invalid-metadata copy.

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/CE-2971

## Review / Merge checklist

- [x] I have seen this code, I have run this code, and I take responsibility for this code.
- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created. 
- [x] Tests included.
- [ ] PR Labeled with `Backport to Beta`, `Backport to Stable`, or `Backport to v1` (if the PR is an urgent fix that needs to be backported)

<a href="https://cubic.dev/pr/n8n-io/n8n/pull/39308?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: packages/cli/src/modules/community-packages/community-packages.service.ts [hunk 1]
```diff
@@ -311,7 +311,7 @@ export class CommunityPackagesService {
 				? `This community node declares an invalid n8n node API version (${JSON.stringify(check.declared)}). Install a version of the package with valid metadata or contact the package author.`
 				: "This community node isn't compatible with your version of n8n. Update n8n to use it.",
 			{
-				requiredNodesApiVersion: isMalformed ? null : Number(check.declared),
+				requiredNodesApiVersion: isMalformed ? null : String(check.declared),
```
Comment (bennycode) [inline packages/cli/src/modules/community-packages/community-packages.service.ts:314]: Since the declared level parsed without errors, could we render the normalized `${major}.${minor}` here and in the two warnings instead of `String(check.declared)`?

**Review by bennycode — APPROVED** [review]
See my comment.
